benchmark/Benchmark.h

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (November of 2022, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <string>
+
+#include "util/Timer.h"
+#include <util/HashMap.h>
+
+/*
+ * Used for measuring the time needed for the execution of a function. It also
+ * saves the measured time togehter with a description of the function
+ * measured.
+ */
+class BenchmarkRecords {
+
+  public:
+
+    // Describes a measured function.
+    struct RecordEntry {
+      std::string descriptor; // Needed, because without it, nobody could tell,
+                              // which time belongs to which benchmark.
+      float measuredTime; // The measured time in seconds.
+    };
+
+    // Describes a group of measured functions.
+    struct RecordGroup {
+      std::string descriptor; // Needed for identifying groups.
+      std::vector<RecordEntry> entries;
+    };
+    
+    // Describes a table of measured functions.
+    struct RecordTable {
+      // For identification.
+      std::string descriptor;
+      // The names of the columns and rows.
+      std::vector<std::string> rowNames;
+      std::vector<std::string> columnNames;
+      // The actual content of the fields. As in the seconds one function
+      // execution takes. Access is [row, column].
+      std::vector<std::vector<float>> entries;
+    };
+
+  private:
+
+    // A vector of all single functions measured.
+    std::vector<RecordEntry> _singleMeasurments;
+
+    // A hash map of all the created RecordGroups. For faster access.
+    // The key for a RecordGroup is it's descriptor.
+    ad_utility::HashMap<std::string, RecordGroup> _recordGroups;
```

Comment:
In all new classes, the `_` should be at the end (we are trying to change that step by step in the existing codebase, so it is

`recordGroups_`, `singleMeasurements_`, `recordTables_` etc.

Reply:
Done, see [2dda05b](https://github.com/ad-freiburg/qlever/pull/860/commits/2dda05beb34836c86f0ccd064c3bf44f310428ad).